diff --git a/homeassistant/bootstrap.py b/homeassistant/bootstrap.py
index f1f1835863b..c1e1500d229 100644
--- a/homeassistant/bootstrap.py
+++ b/homeassistant/bootstrap.py
@@ -105,7 +105,12 @@ from .setup import (
 from .util.async_ import create_eager_task
 from .util.hass_dict import HassKey
 from .util.logging import async_activate_log_queue_handler
-from .util.package import async_get_user_site, is_docker_env, is_virtual_env
+from .util.package import (
+    async_get_user_site,
+    is_docker_env,
+    is_snap_env,
+    is_virtual_env,
+)
 from .util.system_info import is_official_image
 
 with contextlib.suppress(ImportError):
@@ -321,7 +326,7 @@ async def async_setup_hass(
                 err,
             )
         else:
-            if not is_virtual_env():
+            if not is_virtual_env() or is_snap_env():
                 await async_mount_local_lib_path(runtime_config.config_dir)
 
             basic_setup_success = (
diff --git a/homeassistant/helpers/system_info.py b/homeassistant/helpers/system_info.py
index df9679dcb08..975875d0117 100644
--- a/homeassistant/helpers/system_info.py
+++ b/homeassistant/helpers/system_info.py
@@ -11,7 +11,7 @@ from typing import TYPE_CHECKING, Any
 from homeassistant.const import __version__ as current_version
 from homeassistant.core import HomeAssistant
 from homeassistant.loader import bind_hass
-from homeassistant.util.package import is_docker_env, is_virtual_env
+from homeassistant.util.package import is_docker_env, is_snap_env, is_virtual_env
 from homeassistant.util.system_info import is_official_image
 
 from .hassio import is_hassio
@@ -57,6 +57,7 @@ async def async_get_system_info(hass: HomeAssistant) -> dict[str, Any]:
         "virtualenv": is_virtual_env(),
         "python_version": platform.python_version(),
         "docker": False,
+        "snap": False,
         "arch": platform.machine(),
         "timezone": str(hass.config.time_zone),
         "os_name": platform.system(),
@@ -75,6 +76,7 @@ async def async_get_system_info(hass: HomeAssistant) -> dict[str, Any]:
         info_object["os_version"] = await async_get_mac_ver(hass)
     elif platform.system() == "Linux":
         info_object["docker"] = is_docker_env()
+        info_object["snap"] = is_snap_env()
 
     # Determine installation type on current data
     if info_object["docker"]:
@@ -83,6 +85,8 @@ async def async_get_system_info(hass: HomeAssistant) -> dict[str, Any]:
         else:
             info_object["installation_type"] = "Unsupported Third Party Container"
 
+    elif is_snap_env():
+        info_object["installation_type"] = "Home Assistant Snap"
     elif is_virtual_env():
         info_object["installation_type"] = "Home Assistant Core"
 
diff --git a/homeassistant/scripts/__init__.py b/homeassistant/scripts/__init__.py
index f0600b70f48..fca430d40f1 100644
--- a/homeassistant/scripts/__init__.py
+++ b/homeassistant/scripts/__init__.py
@@ -14,7 +14,12 @@ from homeassistant import runner
 from homeassistant.bootstrap import async_mount_local_lib_path
 from homeassistant.config import get_default_config_dir
 from homeassistant.requirements import pip_kwargs
-from homeassistant.util.package import install_package, is_installed, is_virtual_env
+from homeassistant.util.package import (
+    install_package,
+    is_installed,
+    is_snap_env,
+    is_virtual_env,
+)
 
 # mypy: allow-untyped-defs, disallow-any-generics, no-warn-return-any
 
@@ -48,7 +53,7 @@ def run(args: list[str]) -> int:
 
     loop = asyncio.get_event_loop()
 
-    if not is_virtual_env():
+    if not is_virtual_env() or is_snap_env():
         loop.run_until_complete(async_mount_local_lib_path(config_dir))
 
     _pip_kwargs = pip_kwargs(config_dir)
diff --git a/homeassistant/util/package.py b/homeassistant/util/package.py
index 9720bbd4ca3..1b11d59797a 100644
--- a/homeassistant/util/package.py
+++ b/homeassistant/util/package.py
@@ -27,6 +27,9 @@ def is_virtual_env() -> bool:
         sys, "real_prefix"
     )
 
+def is_snap_env() -> bool:
+    """Return if we run in a snap environment."""
+    return os.environ.get("SNAP") is not None
 
 @cache
 def is_docker_env() -> bool:
@@ -134,7 +137,7 @@ def install_package(
         abs_target = os.path.abspath(target)
         args += ["--target", abs_target]
     elif (
-        not is_virtual_env()
+        (not is_virtual_env() or is_snap_env())
         and not (any(var in env for var in _UV_ENV_PYTHON_VARS))
         and (abs_target := site.getusersitepackages())
     ):
